### PR TITLE
Add search endpoint

### PR DESCRIPTION
**Pull request by tbille**
# Summary

Added the search endpoint on the server. It renders a temporary page that diplays a JSON of the searched result. For the moment the search response isn't modified from what I receive from the [snapcraft search API](https://search.apps.ubuntu.com/docs/#id5).

# QA

- `./run`
- http://0.0.0.0:8004/search?q=web&limit=2&offset=60
- Play with the limit and the offset (on the url :smile:) and check that the results are different.

### CODE REVIEW

**Comment by tbille** [comment]
@nottrobin I discussed with @bartaz I will add a commit to filter and normalize the informations I send to the template.

**Comment by tbille** [comment]
I have 2 bugs on the snap search API:
`GET /api/v1/snaps/search?q=web&page=300001&size=2`:
- Returns a snap
- Does not return the last link

Here is an example of what is returned:
```
{
	"_embedded": {
		"clickindex:package": [
			I HAVE 1 SNAP...
		]
	},
	"_links": {
		"first": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?q=web&page=1&size=2"
		},
		"prev": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?q=web&page=44&size=2"
		},
		"self": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?q=web&page=300001&size=2"
		}
	}
}
```

Here is what I would expect:
```
{
	"_links": {
		"first": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?page=1&size=2&q=web"
		},
		"last": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?page=45&size=2&q=web"
		},
		"next": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?page=300002&size=2&q=web"
		},
		"prev": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?page=300000&size=2&q=web"
		},
		"self": {
			"href": "http://search.apps.ubuntu.com/api/v1/snaps/search?q=web&page=300001&size=2"
		}
	}
}
```

The `prev`, `self` at this point are not useful but the `last` argument is needed.

**Comment by nottrobin** [comment]
I got a JSON decode error. I'll keep looking tomorrow.

**Comment by tbille** [comment]
This is the informations that are given to the template:

```
{
  "snaps": [
	{
		"aliases": null,
		"anon_download_url": "https://api.snapcraft.io/api/v1/snaps/download/EQPfyVOJF0AZNz9P2IJ6UKwldLFN5TzS_16.snap",
		"apps": [],
		"architecture": [
			"all"
		],
		"binary_filesize": 12288,
		"channel": "stable",
		"confinement": "strict",
		"contact": "mailto:snappy-devel@lists.ubuntu.com",
		"content": "application",
		"date_published": "2015-03-27T09:50:05.382385Z",
		"deltas": [],
		"description": "This is meant as a fun example for a snappy package.",
		"developer_id": "canonical",
		"developer_name": "Shared snappy store account",
		"download_sha3_384": "cab7ecbfd401b8977ec087435ca46505556db06fb82787532f4ed00fdda9233079b72e25f15e6033369c4314c55e6f2c",
		"download_sha512": "bf2fbe771fcce20377aa76b82fd237382b9e2279ede5d4cc9169095764f520495275acb01a354b5ada119c9612e8f52c63ba998576299dfb474eebdcb656b15f",
		"download_url": "https://api.snapcraft.io/api/v1/snaps/download/EQPfyVOJF0AZNz9P2IJ6UKwldLFN5TzS_16.snap",
		"epoch": "0",
		"icon_url": "https://dashboard.snapcraft.io/site_media/appmedia/2015/03/xkcd.svg.png",
		"last_updated": "2016-04-22T08:58:09.205478+00:00",
		"license": "MIT",
		"name": "xkcd-webserver.canonical",
		"origin": "canonical",
		"package_name": "xkcd-webserver",
		"prices": {},
		"private": false,
		"publisher": "Canonical",
		"ratings_average": 0,
		"release": [
			"16"
		],
		"revision": 16,
		"screenshot_urls": [],
		"snap_id": "EQPfyVOJF0AZNz9P2IJ6UKwldLFN5TzS",
		"summary": "Show random XKCD compic via a build-in webserver",
		"support_url": "mailto:snappy-devel@lists.ubuntu.com",
		"title": "xkcd-webserver",
		"version": "16.04-6"
	},
	...
],
"links": {
	"first": "http://0.0.0.0:8004/search?q=web&limit=2&offset=0",
	"prev": "http://0.0.0.0:8004/search?q=web&limit=2&offset=10",
	"last": "http://0.0.0.0:8004/search?q=web&limit=2&offset=88",
	"self": "http://0.0.0.0:8004/search?q=web&limit=2&offset=12",
	"next": "http://0.0.0.0:8004/search?q=web&limit=2&offset=14"
}
```

File: templates/snap-search.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% extends "_layout.html" %}
+
+{% block title %}{{ snap_title }} — Linux software in the Snap Store{% endblock %}
+
+{% block content %}
```
Comment (nottrobin) [inline templates/snap-search.html]: Could we give the frontends slightly more to work with, with a template something like this?

```
{% block content %}
  <ul>
  {% for snap in snaps %}
    <li>
      <h3>{{ snap.title }}</h1>
      <table>
        {% for key, value in snap.items() %}
          <tr>
            <th>{{ key }}</th>
            <td>{{ value }}</td>
          </tr>
        {% endfor %}
      </table>
    </li>
  {% endfor %}

  <nav>
    {% if links.first %}<a href="{{ links.first }}">First</a>{% endif %}
    {% if links.prev %}<a href="{{ links.prev }}">Previous</a>{% endif %}
    {% if links.next %}<a href="{{ links.next }}">Next</a>{% endif %}
    {% if links.last %}<a href="{{ links.last }}">Last</a>{% endif %}
  </nav>
{% endblock %}
```

File: templates/snap-search.html [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{% extends "_layout.html" %}
```
Comment (nottrobin) [inline templates/snap-search.html]: Do you mind if we call this file simply "search.html"?

**Comment by tbille** [comment]
Basic template added canonical-websites/snapcraft.io#113

**Pull request merged**